shapely/coordinates.py

```diff
@@ -38,19 +52,51 @@ def transform(geometry, transformation, include_z=False):
     >>> transform([Point(0, 0), None], lambda x: x).tolist()
     [<POINT (0 0)>, None]
 
-    By default, the third dimension is ignored:
+    The presence of a third dimension is automatically detected, but can be
+    controlled explicitly:
 
     >>> transform(Point(0, 0, 0), lambda x: x + 1)
-    <POINT (1 1)>
-    >>> transform(Point(0, 0, 0), lambda x: x + 1, include_z=True)
     <POINT Z (1 1 1)>
+    >>> transform(Point(0, 0, 0), lambda x: x + 1, include_z=False)
+    <POINT (1 1)>
+
+    With interleaved=False, the call signature of the transformation is different:
+
+    >>> transform(LineString([(1, 2), (3, 4)]), lambda x, y: (x + 1, y), interleaved=False)
+    <LINESTRING (2 2, 4 4)>
+
+    Or with a z coordinate:
+
+    >>> transform(Point(0, 0, 0), lambda x, y, z: (x + 1, y, z + 2), interleaved=False, include_z=True)
+    <POINT Z (1 0 2)>
+
+    Using pyproj >= 2.1, the following example will reproject Shapely geometries
+    from EPSG 4326 to EPSG 32618:
+
+    >>> from pyproj import Transformer
+    >>> transformer = Transformer.from_crs(4326, 32618, always_xy=True)
+    >>> p = transform(Point(-75, 50), transformer.transform, interleaved=False)
+    >>> shapely.to_wkt(p, rounding_precision=2)
+    'POINT (500000 5538630.7)'
```

Comment:
Is passing this rounding precision manually still needed? (I think with more recent GEOS versions, the default WKT repr tends to be improved, i.e. pruning unnecessary digits more by default)